test/tools/router_check/router.cc

```diff
@@ -0,0 +1,204 @@
+#include "test/tools/router_check/router.h"
+
+Json::ObjectPtr RouterCheckTool::loadJson(const std::string& config_json,
+                                          const std::string& schema) {
+
+  try {
+    // Json configuration
+    Json::ObjectPtr loader = Json::Factory::LoadFromFile(config_json);
+    loader->validateSchema(schema);
+    return loader;
+  } catch (const EnvoyException& ex) {
+    std::cerr << "config schema JSON load failed: " << config_json << std::endl;
+    std::cerr << ex.what() << std::endl;
+    return nullptr;
+  }
+}
+
+bool RouterCheckTool::initializeFromConfig(const std::string& router_config_json) {
+  // TODO(hennna): Allow users to load a full config and extract the route
+  // configuration from it.
+  Json::ObjectPtr loader = loadJson(router_config_json, Json::Schema::ROUTE_CONFIGURATION_SCHEMA);
+
+  if (loader != nullptr) {
```

Comment:
this is an example of error checking you can get rid of if you allow exceptions to bubble up. There is more.